core/src/main/java/hudson/Util.java

```diff
@@ -1415,18 +1422,36 @@ public static Number tryParseNumber(@CheckForNull String numberStr, @CheckForNul
     }
 
     /**
-     * Checks if the public method defined on the base type with the given arguments
-     * are overridden in the given derived type.
+     * Checks if the method defined on the base type with the given arguments
+     * is overridden in the given derived type.
      */
     public static boolean isOverridden(@Nonnull Class base, @Nonnull Class derived, @Nonnull String methodName, @Nonnull Class... types) {
         try {
-            return !base.getMethod(methodName, types).equals(
-                    derived.getMethod(methodName,types));
+            return !getMethod(base, methodName, types).equals(getMethod(derived, methodName, types));
         } catch (NoSuchMethodException e) {
             throw new AssertionError(e);
         }
     }
 
+    private static Method getMethod(@Nonnull Class clazz, @Nonnull String methodName, @Nonnull Class... types) throws NoSuchMethodException {
```

Comment:
Seems it can return null if an exception happens and superclass is null. The method should be marked as CheckForNull